.github/workflows/testing.yml

```diff
@@ -0,0 +1,37 @@
+name: SSE CI/CD
+on: [push]
+
+jobs:
+  style_check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install clang-format
+        run: sudo apt-get install clang-format
+      - name: Run style check
+        run: clang-format -n -Werror src/* tests/*.cpp
+    
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install packages
+        run: sudo apt-get install cmake libyaml-cpp-dev libeigen3-dev libboost-test-dev
+      - name: Run CMake
+        run: mkdir build && cd build && cmake ..
+      - name: Run make
+        run: cd build && make
+
+  test:
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install packages
+        run: sudo apt-get install cmake libyaml-cpp-dev libeigen3-dev libboost-test-dev
+      - name: Run CMake
+        run: mkdir build && cd build && cmake ..
```

Comment:
Here, you build again. For larger projects, this might not be what we want. We could instead reuse the build from the `build` job through an artifact.